engine/updategraph/src/main/java/io/deephaven/engine/liveness/IsLivenessNode.java

```diff
@@ -0,0 +1,40 @@
+package io.deephaven.engine.liveness;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.ref.WeakReference;
+import java.util.stream.Stream;
+
+public interface IsLivenessNode extends LivenessNode {
+    LivenessNode asLivenessNode();
```

Comment:
Should we have some JavaDoc for the class and the `asLivenessNode` method? Mainly, it should make it clear that these things have tightly coupled lifecycles, and that the `LivenessNode` returned must effectively be a proxy for `this`.